### PR TITLE
Milestone5

### PR DESCRIPTION
**Pull request by ccalvarez**

### CODE REVIEW

File: models/event.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+const Schema = mongoose.Schema;
+const eventSchema = new Schema(
+  {
+    _id: { type: Number, required: true },
+    title: { type: String, required: true, index: true},
+    description: { type: String, required: true },
+    date: { type: Date }
+  },
+  {
+    versionKey: false
```
Comment (arielcamus) [inline models/event.js:29]: Why did you add this line here?

File: models/event.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  }
+);
+
+Event = mongoose.model('Event', eventSchema);
+
+Event.ensureIndexes((err) => {
```
Comment (arielcamus) [inline models/event.js:35]: Why do you need this here?

File: routes/events.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  eventManager.getEventsByTitle(titleBuscado)
+  .then(
+    events => res.json(events)
+  ).catch(
+    error => res.status(500).send('Se encontró un error ' + error)
+  )
+});
+
+router.post('/', (req,res) => {
+  eventManager.createEvent(req.body.id, req.body.title, req.body.description, req.body.date)
```
Comment (arielcamus) [inline routes/events.js:42]: It's much simpler if you let Mongo autogenerate the IDs so you don't need to worry about it.

File: routes/eventsManager.js [hunk 4]
```diff
@@ -0,0 +1,167 @@
+const EventModel = require('../models/event');
+
+function getAll(){
+    return new Promise( function (resolve, reject) {
```
Comment (arielcamus) [inline routes/eventsManager.js:4]: You don't really need to create a Promise here.

File: routes/eventsManager.js [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+      reject(ex);
+    }
+  });
+}
+
+function updateEvent(_id, title, description, date) {
+  return new Promise( (resolve, reject) => {
+    try {
+      EventModel.findById(Number(_id), (err, event) => {
+        if(err){
```
Comment (arielcamus) [inline routes/eventsManager.js:101]: You can clean your code by refactoring conditional statements like this to one line.

File: models/event.js (hunk 1 above)
Comment (ccalvarez) [reply models/event.js:29]: Hi Ariel, we added it to remove the field "__v" from documents, because we thought we might not need that field, and also I saw the use of that sentence in the [article](https://scotch.io/tutorials/test-a-node-restful-api-with-mocha-and-chai) suggested in Task # 5 of Milestone 4: Testing your API.

Now I'm reading in [this guide](http://mongoosejs.com/docs/guide.html) that it's not advisable to disable versioning.

File: models/event.js (hunk 2 above)
Comment (ccalvarez) [reply models/event.js:35]: We added this line because we found that without it the index was not created in the collection after our application execution.  We checked the index creation in the mongo shell using the command `db.collection.getIndexes()`.  After adding the .ensureIndexes() call we could see in mongo shell the index created.  I'm not sure if this is needed, or if  Mongoose ensures the use of the index even if it is not persisted.

File: routes/events.js (hunk 3 above)
Comment (ccalvarez) [reply routes/events.js:42]: Thank you, we'll test the fix and commit it to this PR.  How could we prevent the insertion of duplicate events?  I was thinking of a unique index on title + date.

File: models/event.js (hunk 1 above)
Comment (ccalvarez) [reply models/event.js:29]: Fixed with b00d6a0709c81e6504ba279c861cff2df77db149
